lib/license_finder/package_managers/yarn.rb

```diff
@@ -83,6 +92,16 @@ def yarn2_project?
       end
     end
 
+    def yarn3_project?
```

Comment:
This should be refactored to just return the project type. Duplicating the same code and then checking for a number isnt great

Reply:
refactored